manifests/instance.pp

```diff
@@ -0,0 +1,439 @@
+    require => File['/etc/stunnel']
+  }
+
+  if $pki {
+    pki::copy { "stunnel_${_safe_name}":
+      source => $app_pki_external_source,
+      pki    => $pki,
+      notify =>  Service["stunnel_${_safe_name}"]
+    }
+  }
+
+  # NOTE: The pidfile directory is ensured by the service file(s)
+  if $_chroot {
+    file { $_chroot:
+      ensure => 'directory',
+      owner  => 'root',
+      group  => $setgid,
+      mode   => '0770',
+    }
+
+    # The following two entries are required to be able to properly resolve
+    # hosts using the _chroot directory.
+    file { "${_chroot}/etc":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0755',
+    }
+
+    file { "${_chroot}/etc/resolv.conf":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/resolv.conf',
+    }
+
+    file { "${_chroot}/etc/nsswitch.conf":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/nsswitch.conf',
+    }
+
+    file { "${_chroot}/etc/hosts":
+      ensure => 'file',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      source => 'file:///etc/hosts',
+    }
+
+    file { "${_chroot}/var":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644'
+    }
+
+    file { "${_chroot}/etc/pki":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => $setgid,
+      mode   => '0640'
+    }
+
+    $_require_pki =  $pki ? { true =>  Pki::Copy["stunnel_${_safe_name}"], default =>  undef }
+
+    file { "${_chroot}/etc/pki/cacerts":
+      source  => "file://${app_pki_dir}/cacerts",
+      group   => $setgid,
+      mode    => '0640',
+      recurse => true,
+      require =>  $_require_pki
+    }
+  }
+
+  # The rules are pulled together from the accept_* and connect_*
+  # variables.
+  #
+  # This is only enabled if the system is a server.
+  if $firewall and !$client {
+    include '::iptables'
+
+    $_dport = [to_integer(split(to_string($accept),':')[-1])]
+
+    iptables::listen::tcp_stateful { "allow_stunnel_${_safe_name}":
+      trusted_nets => $trusted_nets,
+      dports       => $_dport
+    }
+  }
+
+  if !$client and $tcpwrappers {
+    include '::tcpwrappers'
+
+    tcpwrappers::allow { "allow_stunnel_${_safe_name}":
+      pattern => nets2ddq($trusted_nets)
+    }
+  }
+
+  if 'upstart' in $facts['init_systems'] {
+    $_service_file = "/etc/rc.d/init.d/stunnel_${_safe_name}"
+    file { $_service_file:
+      ensure  => 'present',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0750',
+      content => template('stunnel/instance_init.erb')
+    }
+  }
+  elsif 'systemd' in $facts['init_systems'] {
+    $_service_file = "/etc/systemd/system/stunnel_${_safe_name}.service"
+    file { $_service_file:
```

Comment:
You need to call `systemctl daemon-reload` in order for the system to see this service. The systemd module takes care of this for you. https://github.com/simp/puppet-systemd